### ISSUE TITLE
Integrate visual marks into signatures.vim

### ISSUE THREAD

**Issue by hallzy** [iago-lito/vim-visualMarks#7]
Check out [this plugin](https://github.com/kshenoy/vim-signature).

It puts a letter in the left column showing you where the marks are located. I thought this was kind of nifty, and I think it would be cool if we could add onto this plugin to make it also work with our visual marks.

We could then push our additions to that guys repo, and we could have a note or something in the readme saying that signature supports this visual marks plugin.

**Comment by iago-lito** [comment]
Hey, cool. Why not? Would this mean that I shouldn't worry too much about the form of the documentation yet? I guess it'd just be integrated into signature's documentation. I'll open an issue to them now, asking them whether they are interested.

**Comment by hallzy** [comment]
I think what we should do is continue with what we are doing here. It is possible that people may want this, but not the signatures as well. We could try and get the signatures plugin to work with this one, so call the functions from this plugin if a specific flag is set.

If the guy is not interested, maybe we can still add it to this plugin as an optional feature. Either way, I think it would be beneficial to continue this. If the guy says sure, then we can start an integration, which may require some additions on this plugin, as well as theirs, add some flags (default off because maybe someone wants the signatures but not for visual marks), then when we are done with it, add a note to our readme and the other guys readme, basically saying that the two are compatible of you use x,y, z etc flags.

I have not actually looked at how easy or hard it may be to integrate, so I dont know what the eta would be but I think it is worth looking into.

**Comment by hallzy** [comment]
I created a forked repo of it where we can work on https://github.com/hallzy/vim-signature

I have added you as a collaborator so that you should be able to make changes without needing to do pull requests (This is the first time I have added someone as a collaborator to a repo, so if you have problems let me know)

**Comment by iago-lito** [comment]
Okay, great. I had problems, but I guess it is just because I naively tried to push our current work into `hallzy/vim-signature`, which makes no sense now I'm thinking on it. If I understand well, we will use this forked repo just as a basis for integration into signature: trying to delve into their work, test stuff, hack a little bit, understand etc., then only integrate visualMarks if we can. This is kind of a different job, but I'm enthusiastic!

I haven't looked either wether it'll be difficult or not, so I'm okay to continue yet with what we have done so far. I'm still trying to grasp the basics of Vim's documentation, but I'm also moving for the summer, which is kind of time-consuming. I'll keep you updated :)

**Comment by hallzy** [comment]
My plan was to never actually put the visual marking plugin into this fork. I was thinking that there could be a check to see if the person has visualMarks, or if the user has a flag set in their vimrc to use the visual marks, and if it is set we can use this plugin combined with ours to interact with each other as separate pieces. That way they are still independent so that no one is forced to use both.

Not sure how feasible it will be but that is what I had in mind anyways.

So basically, edit our script, and edit this script so that they will work together when we ask them too.
